docs/developer/payments.mdx

```diff
@@ -8,12 +8,301 @@ The process below describes the key milestones in the Saleor payment flow.
+
+The [transactionProcess](api-reference/payments/mutations/transaction-process.mdx) mutation requires the following arguments:
+
+- `id` - The ID of the transaction returned by [transactionInitialize](api-reference/payments/mutations/transaction-initialize.mdx).
+- `data` - The data to be passed to the payment gateway.
+
+The [transactionProcess](api-reference/payments/mutations/transaction-process.mdx) mutation returns:
+
+- `transaction` - The processed transaction.
+- `transactionEvent` - The event created based on the response received from the payment app.
+- `data` - The data returned by the app.
+- `errors` - Errors related to the mutation call.
+
+The mutation triggers the [TRANSACTION_PROCESS_SESSION](api-reference/webhooks/enums/webhook-event-type-sync-enum.mdx#code-style-fontweight-normal-webhookeventtypesyncenumbtransaction_process_sessionbcode) webhook.
+For more details about this webhook, refer to its [documentation](developer/extending/apps/synchronous-webhooks/transaction-webhooks.mdx#process-transaction-session).
+
+The following example demonstrates how to use the [transactionProcess](api-reference/payments/mutations/transaction-process.mdx) mutation to process the payment:
+
+```graphql
+mutation TransactionInitialize {
+  transactionProcess(
+    id: "VHJhbnNhY3Rpb25JdGVtOjhhMzE1ODVlLWE1MTktNDVkMS1hNTFmLTYwZGZlNGRkM2FkNg"
+    data: { additional: { actions: "details" } }
+  ) {
+    transaction {
+      id
+    }
+    transactionEvent {
+      type
+      pspReference
+    }
+    errors {
+      field
+      message
+      code
+    }
+  }
+}
+```
+
+The response would look like this:
+
+```json
+{
+  "data": {
+    "transactionProcess": {
+      "data": {
+        "some-json": "data"
+      },
+      "transaction": {
+        "id": "VHJhbnNhY3Rpb25JdGVtOjhhMzE1ODVlLWE1MTktNDVkMS1hNTFmLTYwZGZlNGRkM2FkNg"
+      },
+      "transactionEvent": {
+        "type": "CHARGE_SUCCESS",
+        "pspReference": "ppp-123"
+      },
+      "errors": []
+    }
+  }
+}
+```
+
+The current status of a payment can be determined based on the value of `transactionEvent.type`:
+
+- `AUTHORIZATION_ACTION_REQUIRED` or `CHARGE_ACTION_REQUIRED` indicates that additional action is required from the customer to
+  finalize the payment. Calling [transactionProcess](api-reference/payments/mutations/transaction-process.mdx) will be required to pass the result of the action to the app.
+- `AUTHORIZATION_SUCCESS` indicates that funds have been authorized.
+- `AUTHORIZATION_FAILURE` indicates that authorization has failed.
+- `AUTHORIZATION_REQUEST` indicates that authorization is in a pending state.
+- `CHARGE_SUCCESS` indicates that funds have been charged.
+- `CHARGE_FAILURE` indicates that charging has failed.
+- `CHARGE_REQUEST` indicates that charging is in a pending state.
+
+## Saleor App with existing Checkout App
```

Comment:
The part about `transactionEventReport` from this section also applies to Payment Apps. We use `transactionEventReport` to react to Adyen webhooks for example.

Reply:
right. I need to rebuild this page. Thanks!